Expand and consolidate networking docs (#68051)
elastic/elasticsearch: docs/reference/modules/transport.asciidoc

[[transport-settings]]
==== Transport settings

> The following settings allow you to configure the transport interface
> independently of the <<http-settings,HTTP interface>>. You can also configure
> both interfaces together using the <<common-network-settings,network
> settings>>.

Reviewer: ```suggestion
Use the following settings to configure the transport interface
independently of the <<http-settings,HTTP interface>>. Use the
<<common-network-settings,network
settings>> to configure both interfaces together.
```

`transport.port`::
(<<static-cluster-setting,Static>>)
The port to bind for communication between nodes. Accepts a single value or a
range. If a range is specified, the node will bind to the first available port

> in the range. You should set this setting to a single port, not a range, on
> every master-eligible node.

Reviewer: ```suggestion
in the range. Configure this setting to use a single port only on
every master-eligible node.
```

+
Defaults to `9300-9400`.

`transport.host`::
(<<static-cluster-setting,Static>>)
Sets the address of this node for transport traffic. The node will bind to this
address and will also use it as its transport publish address. Accepts an IP
address, a hostname, or a <<network-interface-values,special value>>. You
should only use this setting if you require different configurations for the
transport and HTTP interfaces.
+
Defaults to the address given by `network.host`.

`transport.bind_host`::
(<<static-cluster-setting,Static>>)

> The network address(es) to which the node should bind in order to listen for
> incoming transport connections. Accepts a list of IP addresses, hostnames, and

Reviewer: ```suggestion
Bind addresses the node should use to listen for incoming
transport connections. Accepts a list of IP addresses, hostnames, and
```

<<network-interface-values,special values>>. Defaults to the address given by
`transport.host` or `network.bind_host`. You should only use this setting if
you wish to bind to multiple addresses or to use different addresses for
publishing and binding, and you require different binding configurations for
the transport and HTTP interfaces.

`transport.publish_host`::
(<<static-cluster-setting,Static>>)
The network address at which the node can be contacted by other nodes. Accepts
an IP address, a hostname, or a <<network-interface-values,special value>>.
Defaults to the address given by `transport.host` or `network.publish_host`.

> You should only use this setting if you wish to bind to multiple addresses or
> to use different addresses for publishing and binding, and you require
> different publishing configurations for the transport and HTTP interfaces.

Reviewer: ```suggestion
Use this setting only if you require
different binding configurations for the transport or HTTP interfaces, 
such as binding to multiple addresses or using different addresses for publishing
and binding.
```

`transport.publish_port`::
(<<static-cluster-setting,Static>>)
The port of the <<modules-network-binding-publishing,transport publish

> address>>. You should only set this if you need the publish port to be
> different from `transport.port`. Defaults to the port assigned via
> `transport.port`.

Reviewer: ```suggestion
address>>. Set this parameter only if you need the publish port to be
different from `transport.port`. Defaults to the port assigned via
`transport.port`.
```
